ml-algorithms/src/main/java/org/opensearch/ml/engine/encryptor/EncryptorImpl.java

```diff
@@ -121,6 +121,18 @@ public String generateMasterKey() {
         return Base64.getEncoder().encodeToString(keyBytes);
     }
 
+    // Refresh the key from the config index.
+    public void refreshMasterKey(ActionListener<Boolean> listener) {
+        try {
+            // Currently, we only handle no tenant case.
```

Comment:
Is that because the tenantId is not available when running the syncUp on each node? otherwise just fill in the tenantId in the initMasterKeyFromIndex(tenantId) it should refresh the masterKey.